tests/integration/crunchers/test_cruncher_equal_quantile_walk.py

```diff
@@ -0,0 +1,351 @@
+        tcruncher = self.tclass(test_db)
+        lead = ["Emissions|CO2"]
+        follow = "Emissions|CH4"
+        simple_df = self._adjust_time_style_to_match(simple_df, test_db)
+        res = tcruncher.derive_relationship(follow, lead)
+        if add_col:
+            add_col_val = "blah"
+            simple_df[add_col] = add_col_val
+            simple_df = IamDataFrame(simple_df.data)
+            assert simple_df.extra_cols[0] == add_col
+
+        infilled = res(simple_df)
+        # We compare the results with the expected results: for T1, we are below the
+        # lower limit on the first, in the middle on the second. At later times we are
+        # always above the highest value.
+        time_filter = {infilled.time_col: [infilled[infilled.time_col][0]]}
+        assert np.allclose(infilled.filter(**time_filter)["value"].values, [50, 100])
+        for time_ind in range(1, 3):
+            time_filter = {infilled.time_col: [infilled[infilled.time_col][time_ind]]}
+            assert np.allclose(
+                infilled.filter(**time_filter)["value"].values,
+                max(
+                    test_db.filter(variable=follow)
+                    .filter(**time_filter)["value"]
+                    .values
+                ),
+            )
+
+        # Test we can append our answer
+        append_df = simple_df.filter(variable=lead).append(infilled)
+        assert append_df.filter(variable=follow).equals(infilled)
+
+        if add_col:
+            assert all(append_df[add_col] == add_col_val)
+
+    def test_numerical_relationship(self, test_db):
+        # Calculate the values using the cruncher for a fairly detailed dataset
+        large_db = IamDataFrame(self.large_db.copy())
+        large_db = self._adjust_time_style_to_match(large_db, test_db,)
+        large_db["value"] -= 0.1
+        test_db.filter(year=2010, inplace=True)
+        tcruncher = self.tclass(large_db)
+        lead = ["Emissions|CO2"]
+        follow = "Emissions|CH4"
+        res = tcruncher.derive_relationship(follow, lead)
+        assert callable(res)
+        to_find = test_db
+        crunched = res(to_find)
+
+        # Calculate the same values numerically
+        xs = np.sort(large_db.filter(variable="Emissions|CO2")["value"].values)
+        ys = np.sort(large_db.filter(variable="Emissions|CH4")["value"].values)
+        quantiles_of_x = np.arange(len(xs)) / (len(xs) - 1)
+        quant_of_y = scipy.interpolate.interp1d(
+            xs, quantiles_of_x, bounds_error=False, fill_value=(0, 1)
+        )(to_find.filter(variable=lead)["value"].values)
+
+        expected = np.quantile(ys, quant_of_y)
+
+        assert all(crunched["value"].values == expected)
+
+    def test_with_one_value_in_infiller_db(self, test_db, caplog):
+        # The calculation is different with only one entry in the infiller db. We
+        # expect a warning and the only value to be returned in all cases.
```

Comment:
```suggestion
        # 
```

No warning at the moment ?

Reply:
True. Probably not going to add one now, they are annoying